### ISSUE TITLE
BUG: KASAN: slab-out-of-bounds in jffs2_rtime_compress

### ISSUE THREAD

**Issue by shenki** [openbmc/linux#151]
```
[  232.340401] ==================================================================
[  232.347764] BUG: KASAN: slab-out-of-bounds in jffs2_rtime_compress+0x228/0x424
[  232.355048] Write of size 1 at addr b3a37c01 by task phosphor-timema/1050
[  232.361859] 
[  232.363407] CPU: 0 PID: 1050 Comm: phosphor-timema Not tainted 4.13.16-01407-g20e4481340b1 #59
[  232.372041] Hardware name: Generic DT based system
[  232.376955] [<8010edb4>] (unwind_backtrace) from [<8010aa70>] (show_stack+0x20/0x24)
[  232.384815] [<8010aa70>] (show_stack) from [<80a992d8>] (dump_stack+0x20/0x28)
[  232.392172] [<80a992d8>] (dump_stack) from [<8030f248>] (print_address_description+0x5c/0x32c)
[  232.400902] [<8030f248>] (print_address_description) from [<8030f808>] (kasan_report+0x14c/0x3)
[  232.409881] [<8030f808>] (kasan_report) from [<8030e9e4>] (__asan_store1+0x44/0x48)
[  232.417651] [<8030e9e4>] (__asan_store1) from [<8041e3a4>] (jffs2_rtime_compress+0x228/0x424)
[  232.426301] [<8041e3a4>] (jffs2_rtime_compress) from [<803fa7bc>] (jffs2_selected_compress+0xf)
[  232.435722] [<803fa7bc>] (jffs2_selected_compress) from [<803fa9fc>] (jffs2_compress+0x60/0x57)
[  232.444612] [<803fa9fc>] (jffs2_compress) from [<80409934>] (jffs2_write_inode_range+0x1a4/0x5)
[  232.453590] [<80409934>] (jffs2_write_inode_range) from [<803fe498>] (jffs2_write_end+0x308/0x)
[  232.462678] [<803fe498>] (jffs2_write_end) from [<8029355c>] (generic_perform_write+0x1e4/0x3c)
[  232.471573] [<8029355c>] (generic_perform_write) from [<80293978>] (__generic_file_write_iter+)
[  232.481330] [<80293978>] (__generic_file_write_iter) from [<80293f84>] (generic_file_write_ite)
[  232.491262] [<80293f84>] (generic_file_write_iter) from [<8031c47c>] (__vfs_write+0x1f4/0x314)
[  232.499966] [<8031c47c>] (__vfs_write) from [<8031c8a0>] (vfs_write+0xec/0x2f4)
[  232.507365] [<8031c8a0>] (vfs_write) from [<8031cbd8>] (SyS_write+0x64/0xcc)
[  232.514512] [<8031cbd8>] (SyS_write) from [<80103be0>] (ret_fast_syscall+0x0/0x3c)
[  232.522113] 
[  232.523641] Allocated by task 1050:
[  232.527204]  kasan_kmalloc+0xd4/0x174
[  232.530929]  __kmalloc+0xe4/0x254
[  232.534323]  jffs2_selected_compress+0x48/0x2d8
[  232.538923]  jffs2_compress+0x60/0x578
[  232.542733]  jffs2_write_inode_range+0x1a4/0x5e8
[  232.547416]  jffs2_write_end+0x308/0x700
[  232.551410]  generic_perform_write+0x1e4/0x3c4
[  232.555916]  __generic_file_write_iter+0x23c/0x568
[  232.560770]  generic_file_write_iter+0x2e0/0x570
[  232.565447]  __vfs_write+0x1f4/0x314
[  232.569084]  vfs_write+0xec/0x2f4
[  232.572453]  SyS_write+0x64/0xcc
[  232.575740]  ret_fast_syscall+0x0/0x3c
[  232.579511] 
[  232.581033] Freed by task 0:
[  232.583931] (stack is not available)
[  232.587526] 
[  232.589054] The buggy address belongs to the object at b3a37c00
[  232.589054]  which belongs to the cache kmalloc-32 of size 32
[  232.600750] The buggy address is located 1 bytes inside of
[  232.600750]  32-byte region [b3a37c00, b3a37c20)
[  232.610861] The buggy address belongs to the page:
[  232.615698] page:be89e6e0 count:1 mapcount:0 mapping:b3a37000 index:0xb3a37fc1
[  232.622975] flags: 0x100(slab)
[  232.626115] raw: 00000100 b3a37000 b3a37fc1 0000003f 00000001 be85f294 b4801328 b4800da0
[  232.634235] page dumped because: kasan: bad access detected
[  232.639829] 
[  232.641338] Memory state around the buggy address:
[  232.646180]  b3a37b00: fb fb fb fb fc fc fc fc fb fb fb fb fc fc fc fc
[  232.652757]  b3a37b80: fb fb fb fb fc fc fc fc fb fb fb fb fc fc fc fc
[  232.659331] >b3a37c00: 01 fc fc fc fc fc fc fc 00 00 04 fc fc fc fc fc
[  232.665889]            ^
[  232.668473]  b3a37c80: 00 00 04 fc fc fc fc fc fb fb fb fb fc fc fc fc
[  232.675054]  b3a37d00: fb fb fb fb fc fc fc fc fb fb fb fb fc fc fc fc
[  232.681611] ==================================================================
```

**Comment by shenki** [comment]
```
fs/jffs2/compr_rtime.c:58
```

**Comment by shenki** [comment]
This was fixed in mainline by 90ada91f4610c5ef11bc52576516d96c496fc3f1. This has not been backported at this stage.

**Comment by shenki** [comment]
The patch has been applied to the stable trees and will appear in the openbmc tree when they are merged (eg v5.10.36)
